src/Opw.PineBlog.EntityFrameworkCore/BlogEntityDbContext.cs

```diff
@@ -1,10 +1,18 @@
 using Microsoft.EntityFrameworkCore;
 using Opw.PineBlog.Entities;
 using Opw.EntityFrameworkCore;
+using System.Threading.Tasks;
+using System;
+using Opw.PineBlog.Models;
+using System.Threading;
+using System.Linq;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Opw.HttpExceptions;
 
 namespace Opw.PineBlog.EntityFrameworkCore
 {
-    public class BlogEntityDbContext : EntityDbContext, IBlogEntityDbContext
+    public class BlogEntityDbContext : EntityDbContext, IRepository
```

Comment:
The idea of a DbContext in EF is that is acts as a unit of work, and its DbSets are the repositories. So having now one repository breaks that principle. Create repositories per DbSet and a unit of work for the DbContext.

Reply:
Are you envisioning like so:
`IAuthorRepository`,
`IPostRepository`, etc. then the `BlogEntityDbContext` would be an implementation of both?

Reply:
Yes I would like to have a separate `UnitOfWork` class and separate `Repository` classes, that way the `BlogEntityDbContext` would remain pure